sudo/src/run_handler.rs

```diff
@@ -191,6 +191,53 @@ pub fn run_target(
     do_request(req, copy_env, manually_requested_dir)
 }
 
+/// Returns all the extensions from PATHEXT that are used for "executables"
+fn get_path_extensions() -> Vec<String> {
+    env::var("PATHEXT")
+        .unwrap_or_else(|_| {
+            // If PATHEXT isn't set, use the default
+            ".COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC".to_string()
+        })
+        .split(';')
+        .map(|ext| ext.to_string())
+        .collect()
+}
+
+/// Searches the PATH for the given filename, with the extensions from PATHEXT.
+/// Returns the full path to the file if it's found, or an error if it's not.
+fn search_path_with_extensions(filename: &str) -> Result<String> {
+    let filename = HSTRING::from(filename);
+    let mut buffer = vec![0; 260];
+    let extensions = get_path_extensions();
+    for extension in extensions {
+        let extension_hstring = HSTRING::from(extension);
+        // SearchPathW will ignore the extension if the filename already has one.
+        let len: u32 =
+            unsafe { SearchPathW(None, &filename, &extension_hstring, Some(&mut buffer), None) };
+
+        // If the function fails, the return value is zero.
+        if len == 0 {
+            continue;
+        }
+
+        // If the function succeeds, the value returned is the
+        // length of the string that is copied to the buffer,
+        // not including the terminating null character.
+        if len < buffer.len() as u32 {
+            buffer.truncate(len as usize);
+            return Ok(String::from_utf16(&buffer)?);
+        }
+
+        // If the return value is greater than nBufferLength, the value
+        // returned is the size of the buffer that is required to hold
+        // the path, including the terminating null character.
+        //
+        // Includes some padding just in case and because it's cheap.
+        buffer.resize((len + 64) as usize, 0);
```

Comment:
Wait, doesn't this mean that if it fails to fit in the buffer _we will move on to the next extension_ after growing the buffer? That doesn't seem correct.

Reply:
Hah. I didn't even catch this! My initial suggestion had a loop, so this looked quite similar. 😅 But it needs a loop in a loop.